cmd/talosctl/cmd/mgmt/gen/config.go

```diff
@@ -73,28 +73,31 @@ var genConfigCmdFlags struct {
 	withSecrets             string
 }
 
-// genConfigCmd represents the `gen config` command.
-var genConfigCmd = &cobra.Command{
-	Use:   "config <cluster name> <cluster endpoint>",
-	Short: "Generates a set of configuration files for Talos cluster",
-	Long: `The cluster endpoint is the URL for the Kubernetes API. If you decide to use
+// NewConfigCmd builds the config generation subcommand with the given name.
+func NewConfigCmd(name string) *cobra.Command {
+	return &cobra.Command{
+		Use:   fmt.Sprintf("%s <cluster name> <cluster endpoint>", name),
+		Short: "Generates a set of configuration files for Talos cluster",
+		Long: `The cluster endpoint is the URL for the Kubernetes API. If you decide to use
 a control plane node, common in a single node control plane setup, use port 6443 as
 this is the port that the API server binds to on every control plane node. For an HA
 setup, usually involving a load balancer, use the IP and port of the load balancer.`,
-	Args: cobra.ExactArgs(2),
-	RunE: func(cmd *cobra.Command, args []string) error {
-		err := validateClusterEndpoint(args[1])
-		if err != nil {
-			return err
-		}
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := validateClusterEndpoint(args[1])
+			if err != nil {
+				return err
+			}
 
-		switch genConfigCmdFlags.configVersion {
-		case "v1alpha1":
-			return writeV1Alpha1Config(args)
-		}
+			//nolint:gocritic
+			switch genConfigCmdFlags.configVersion {
+			case "v1alpha1":
+				return writeV1Alpha1Config(args)
+			}
```

Comment:
Maybe we should return an error if `configVersion` string was incorrect?

Reply:
good point, added that